cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/ViewSystemMetadataWriter.java

```diff
@@ -57,6 +63,20 @@ public ViewSystemMetadataWriter(MetadataStore metadataStore, Id.Stream.View view
   @Override
   protected String getSchemaToAdd() {
     Schema schema = viewSpec.getFormat().getSchema();
+    if (schema == null) {
+      FormatSpecification format = viewSpec.getFormat();
+      RecordFormat<Object, Object> initializedFormat = null;
+      try {
+        initializedFormat = RecordFormats.createInitializedFormat(format);
+      } catch (IllegalAccessException | InstantiationException | UnsupportedTypeException | ClassNotFoundException e) {
+        LOG.debug("Exception: ", e);
+        LOG.warn("Exception while determining schema for view {}. View {} will not contain schema as metadata.", viewId,
+                 viewId);
+      }
+      if (initializedFormat != null) {
```

Comment:
its initialized to null on line 70 and we suppress exceptions above, so it can be null here. But we can remove this null check if I move this inside the ``try...catch``